packages/vite/src/node/server/hmr.ts

```diff
@@ -45,6 +45,35 @@ export interface HmrContext {
   server: ViteDevServer
 }
 
+export interface HMRChannel {
+  /**
+   * Broadcast events to all clients
+   */
+  send(payload: HMRPayload): void
+  /**
+   * Send custom event
+   */
+  send<T extends string>(event: T, payload?: InferCustomEventPayload<T>): void
+  /**
+   * Handle custom event emitted by `import.meta.hot.send`
+   */
+  on<T extends string>(
+    event: T,
+    listener: (data: InferCustomEventPayload<T>, ...args: any[]) => void,
+  ): void
+  /**
+   * Unregister event listener.
+   */
+  off(event: string, listener: Function): void
+  close(): void
+}
+
+// TODO: more jsdoc
+export interface HMRBroadcaster extends HMRChannel {
+  channels: HMRChannel[]
+  addChannel(connection: HMRChannel): void
+}
```

Comment:
Should `channels` be private/internal if we would like to provide methods like `addChannel`? Right now the method only pushes to the Array, so we could just have `channels` and let the user add their channels how they see fit.
`addChannel` could later on do logging, or init stuff though.

Reply:
I made it public only because ws has "clients" public 😄 

To me, this is more of a private method or a read-only getter

Reply:
Made it `readonly` in types, and now return a new array each time (`[...channels]`)